### DISCUSSION TITLE
Add Trigger ID as Trigger condition to Action selector in Blueprint

### DISCUSSION THREAD

**Discussion by theciso** [home-assistant/frontend#15923] [Automations / scripts / scenes]
When using [Trigger ID's](https://www.home-assistant.io/docs/automation/trigger/#trigger-id) in a [Blueprint automation](https://www.home-assistant.io/docs/automation/using_blueprints/), these are not available as a [Trigger condition](https://www.home-assistant.io/docs/scripts/conditions/#trigger-condition) in an [Action selector](https://www.home-assistant.io/docs/blueprint/selectors/#action-selector). However, the Trigger ID's do work when referring to them directly in the YAML of the automation that is using the Blueprint.

` - if:
       - condition: trigger
          id: "<trigger id defined within the blueprint>"
     then:[]`

The Trigger ID's defined in a Blueprint should be made available as a Trigger condition in an Action selector of a Blueprint automation. This would allow users to make use of the GUI instead of having to manually add YAML code to the automation. In addition, this would also provide a consistent user experience with Trigger ID's used in automations without a Blueprint.

**Comment by paulbusse** [comment]
+1

**Comment by karwosts** [comment]
Seems like this might be pretty difficult as frontend doesn't even have this information. The only information core provides about blueprints is their list of user selectors, it doesn't give us any information about what triggers are baked into the automation 😕 . So this would need core work as well, possibly new apis.
